pinot-common/src/main/java/org/apache/pinot/common/utils/DatabaseUtils.java

```diff
@@ -84,4 +84,13 @@ public static String translateTableName(String tableName, HttpHeaders headers, b
   public static String translateTableName(String tableName, HttpHeaders headers) {
     return translateTableName(tableName, headers, false);
   }
+
+  public static boolean isPartOfDatabase(String tableName, @Nullable String databaseName) {
+    // assumes tableName will not have default database prefix ('default.')
+    if (StringUtils.isEmpty(databaseName) || databaseName.equalsIgnoreCase(CommonConstants.DEFAULT_DATABASE)) {
+      return StringUtils.split(tableName, '.').length == 1;
```

Comment:
Shouldn't be better to do `!tableName.contains(".")`?

It has to be faster and doesn't allocate

Reply:
Agree, thanks for catching! Will fix it.